include/picongpu/particles/debyeLength/Estimate.kernel

```diff
@@ -0,0 +1,228 @@
+                 * @tparam T_ElectronBox electron storage type
+                 * @tparam T_Mapping mapping type
+                 * @tparam T_EstimateBox result storage type, device box for Estimate
+                 *
+                 * @param acc alpaka accelerator
+                 * @param electronBox electron storage
+                 * @param mapper mapping
+                 * @param estimateBox device box with a single element to write kernel result to
+                 */
+                template<typename T_Acc, typename T_ElectronBox, typename T_Mapping, typename T_EstimateBox>
+                DINLINE void operator()(
+                    T_Acc const& acc,
+                    T_ElectronBox const electronBox,
+                    T_Mapping mapper,
+                    T_EstimateBox estimateBox) const
+                {
+                    auto const supercellIdx = mapper.getSuperCellIndex(DataSpace<simDim>(cupla::blockIdx(acc)));
+                    auto const& supercell = electronBox.getSuperCell(supercellIdx);
+                    // Supercells with no particles do not contribute to Debye length estimate
+                    if(supercell.getNumParticles())
+                        process(acc, electronBox, supercellIdx, estimateBox);
+                }
+
+            private:
+                /** Check Debye length for electrons in the given supercell
+                 *
+                 * Ths supercell is required to have macroparticles
+                 *
+                 * @tparam T_Acc alpaka accelerator type
+                 * @tparam T_ElectronBox electron storage type
+                 * @tparam T_EstimateBox result storage type, device box for Estimate
+                 *
+                 * @param acc alpaka accelerator
+                 * @param electronBox electron storage
+                 * @param supercellIdx supercell index, including guards
+                 * @param estimate device box with a single element to write kernel result to
+                 */
+                template<typename T_Acc, typename T_ElectronBox, typename T_EstimateBox>
+                DINLINE void process(
+                    T_Acc const& acc,
+                    T_ElectronBox const electronBox,
+                    pmacc::DataSpace<simDim> const supercellIdx,
+                    T_EstimateBox estimateBox) const
+                {
+                    using namespace pmacc::mappings::threads;
+                    constexpr uint32_t frameSize = pmacc::math::CT::volume<SuperCellSize>::type::value;
+                    constexpr uint32_t numWorkers = T_numWorkers;
+                    using ParticleDomCfg = IdxConfig<frameSize, numWorkers>;
+                    /* Estimate variance in momentum of macroparticles in the supercell.
+                     * Each thread processes its particles, then the master thread performs reduction.
+                     * Sums are weighted and operate with single- (not macro-) particle momentums.
```

Comment:
```suggestion
                     * Sums are weighted and operate with single- (not macro-) particle momenta.
```
I like momentums much better since it indicated the plural clearly, but sadly its momenta. 😢